src/main/java/com/puppycrawl/tools/checkstyle/xpath/ElementNode.java

```diff
@@ -100,12 +106,17 @@ private void createChildren() {
      */
     @Override
     public String getAttributeValue(String namespace, String localPart) {
-        if (TEXT_ATTRIBUTE_NAME.equals(localPart)) {
-            return ident;
+        final String result;
+        if (VALUE_ATTRIBUTE_NAME.equals(localPart)) {
+            result = value;
+        }
+        else if (TEXT_ATTRIBUTE_NAME.equals(localPart)) {
+            result = ident;
         }
         else {
             throw throwUnsupportedOperationException();
```

Comment:
Following the documentation of `getAttributeValue(...)` this line should be `return null;` instead.
see http://www.saxonica.com/html/documentation/javadoc/net/sf/saxon/om/NodeInfo.html#getAttributeValue-java.lang.String-java.lang.String-

Reply:
fixed